flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -145,7 +147,7 @@ public MetricRegistryImpl(MetricRegistryConfiguration config, Collection<Reporte
 					} else {
 						LOG.info("Reporting metrics for reporter {} of type {}.", namedReporter, className);
 					}
-					reporters.add(reporterInstance);
+					reporters.add(new ReporterAndSettings(reporterInstance, new ReporterScopedSettings(reporters.size() - 1)));
```

Comment:
nit: If we introduce a static factory method for `ReporterAndSettings`, then we have a single place where instances of this class are being instantiated.

Reply:
I don't think we'll ever need to create this anywhere else.